src/shared/parameter_v2/examples/generated_dynamic_parameters.hpp

```diff
@@ -0,0 +1,351 @@
+            std::const_pointer_cast<const NumericParameter<float>>(example_float_param),
+            std::const_pointer_cast<const Parameter<std::string>>(example_string_param),
+            std::const_pointer_cast<const EnumeratedParameter<RefereeCommand>>(example_enum_param),
+            std::const_pointer_cast<const foo>(foo_config),
+        };
+    }
+
+    const std::shared_ptr<const Parameter<bool>> exampleBoolParam() const
+    {
+        return std::const_pointer_cast<const Parameter<bool>>(example_bool_param);
+    }
+
+    const std::shared_ptr<Parameter<bool>> mutableExampleBoolParam()
+    {
+        return example_bool_param;
+    }
+
+    const std::shared_ptr<const NumericParameter<int>> exampleIntParam() const
+    {
+        return std::const_pointer_cast<const NumericParameter<int>>(example_int_param);
+    }
+
+    const std::shared_ptr<NumericParameter<int>> mutableExampleIntParam()
+    {
+        return example_int_param;
+    }
+
+    const std::shared_ptr<const NumericParameter<uint>> exampleUintParam() const
+    {
+        return std::const_pointer_cast<const NumericParameter<int>>(example_int_param);
+    }
+
+    const std::shared_ptr<NumericParameter<uint>> mutableExampleUintParam()
+    {
+        return example_uint_param;
+    }
+
+    const std::shared_ptr<const NumericParameter<float>> exampleFloatParam() const
+    {
+        return std::const_pointer_cast<const NumericParameter<float>>(example_int_param);
+    }
+
+    const std::shared_ptr<NumericParameter<float>> mutableExampleFloatParam()
+    {
+        return example_float_param;
+    }
+
+    const std::shared_ptr<const Parameter<std::string>> exampleStringParam() const
+    {
+        return std::const_pointer_cast<const NumericParameter<std::string>>(example_string_param);
+    }
+
+    const std::shared_ptr<Parameter<std::string>> mutableExampleStringParam()
+    {
+        return example_string_param;
+    }
+
+    const std::shared_ptr<const EnumeratedParameter<RefereeCommand>> exampleEnumParam() const
+    {
+        return std::const_pointer_cast<const EnumeratedParameter<RefereeCommand>>(example_enum_param);
+    }
+
+    const std::shared_ptr<EnumeratedParameter<RefereeCommand>> mutableExampleEnumParam()
+    {
+        return example_enum_param;
+    }
+
+    const std::shared_ptr<const foo> fooConfig()
+    {
+        return std::const_pointer_cast<const foo>(foo_config);
+    }
+
+    const std::shared_ptr<foo> mutableFooConfig()
+    {
+        return foo_config;
+    }
+
+    const std::string name() const
+    {
+        return "example"
+    }
+
+    bool loadFromCommandLineArguments(int argc, char **argv) {
+        struct commandLineArgs {
+            bool help = false;
+            bool example_bool_param = true;
+            int example_int_param = 3;
+            uint example_uint_param = 3;
+            float example_float_param = 4.04;
```

Comment:
the included configs would also have to get tossed in here, (with the respective boost cli args generated) we might want to share configs between different binaries

Reply:
Can you give an example of how configs might be shared between binaries? Also, not sure if I completely understand what you mean by with the respective cli args generated.

Reply:
`parameter/config/full_system_main_command_line_args.yaml`
`parameter/config/standalone_simulator_main_command_line_args.yaml`
`parameter/config/robot_diagnostics_main_command_line_args.yaml`

All share the networking parameters, this is a small example, but we could put networking stuff in its own config file, and then include it in all the `***main_command_line_args.yaml` to avoid duplicating the definitions like we do now.

If we included the networking config into the example config, the struct would be generated to something like this
~~~
  struct commandLineArgs {
            bool help = false;
            bool example_bool_param = true;
            int example_int_param = 3;
            uint example_uint_param = 3;
            float example_float_param = 4.04;
            // this would be the included config
           struct NetworkingConfig {
                std::string interface;
           }
~~~

and we would have the corresponding boost cli stuff generated
~~~ 
desc.add_options()("interface", boost::program_options::value<std::string>(
&args.NetworkingConfig.interface), "network interface");
~~~

Reply:
Then it seems like we need a rule for these config files that any config parameter names have to be unique across all configs, or at least if any of them have dependencies.

Another option could be to append the config name before it: 
`desc.add_options()("networkingConfig.interface", boost::program_options::value<std::string>(
&args.NetworkingConfig.interface), "network interface");`

Tried the second approach, let me know what you think.

Reply:
Yeah that makes a lot of sense, there is no requirement right now that parameters have to have unique names across multiple configs, so that looks good